docker/DockerBuild.sh

```diff
@@ -133,14 +136,28 @@ if [ -n "${BUILD_NUMBER}" ]; then
   TARGZ_TAG="${TARGZ_TAG}-${BUILD_NUMBER}"
 fi
 
-DOCKER_COMMAND="docker build "
+DOCKER_BUILD_START="docker build "
 BUILD_ARGS="--build-arg UID=${UID_ARG} \
             --build-arg GID=${GID_ARG} \
             --build-arg MINIFI_VERSION=${MINIFI_VERSION} \
             --build-arg DUMP_LOCATION=${DUMP_LOCATION} \
             --build-arg DISTRO_NAME=${DISTRO_NAME} ${BUILD_ARGS}"
 
-DOCKER_COMMAND="${DOCKER_COMMAND} ${BUILD_ARGS} \
+if [ -n "${DOCKER_CCACHE_DUMP_LOCATION}" ]; then
+  DOCKER_COMMAND="${DOCKER_BUILD_START} ${BUILD_ARGS} \
+                -f ${DOCKERFILE} \
+                --target build \
+                -t \
+                minifi_build .."
+  echo "Build image Docker Command: 'DOCKER_BUILDKIT=1 ${DOCKER_COMMAND}'"
+  DOCKER_BUILDKIT=1 ${DOCKER_COMMAND}
+  container_id=$(docker run --rm -d  minifi_build sh -c "while true; do sleep 1; done")
+  mkdir -p "${DOCKER_CCACHE_DUMP_LOCATION}"
+  docker cp "${container_id}:/home/minificpp/.ccache/." "${DOCKER_CCACHE_DUMP_LOCATION}"
```

Comment:
Shouldn't we volume mount the host $HOME/.ccache to the container instead? We could avoid the sleep hack and the copy this way.

Reply:
Unfortunately there is no way to mount a host directory in build time (it's due to a principle that it would ruin the reproducibility of the image). The only way that would be possible is to create a build image, and run the build commands in the container while mounting the host ccache. But that would also mean we could not use multi-stage builds, so the built binaries would have to be copied out manually and added to the final image separately.

Reply:
The problem with mounting is that is causes permission errors, because the userid of the host differ from the userid used in the container and if we mount the directory the files are copied to mounted directory with the userid and groupid of the container user (and we would also have to change the permissions of the directory to be able to copy there). With `docker cp` we don't have this problem.